### PR TITLE
Run `forc fmt` on  std-lib

### PR DESCRIPTION
**Pull request by crodas**
## Description

Run `forc fmt` on the std-lib

## Checklist

- [x] I have linked to any relevant issues.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

File: sway-lib-std/src/address.sw [hunk 1]
```diff
@@ -106,8 +106,9 @@ impl Address {
         while index < number_of_outputs {
             if let Output::Variable = output_type(index) {
                 if output_amount(index) == 0 {
-                    asm(r1: self.value, r2: index, r3: amount, r4: asset_id.value) {
-                        tro r1 r2 r3 r4;
+                    asm(r1: self.value, r2: index, r3: amount, r4: asset_id
+                        .value) {
```
Comment (sdankel) [inline sway-lib-std/src/address.sw:110]: This line break doesn't look right. I think it should either stay the same or look like this:

```
                    asm(
                        r1: self.value,
                        r2: index,
                        r3: amount,
                        r4: asset_id.value
                    ) {
```

File: test/src/e2e_vm_tests/test_programs/should_fail/trait_cannot_find_in_scope_issue/test.toml [hunk 2]
```diff
@@ -4,7 +4,7 @@ category = "fail"
 #check: $()let _ = sha256(123u8);
 #nextln: $()Trait "Hash" is not imported into "main.sw" when calling "sha256".
 #check: $()Import the "Hash" trait into "main.sw" by using: `use std::hash::Hash;`.
-#check: $()pub fn sha256<T>(s: T) -> b256 where T: Hash {
+#check: $()T: Hash,
```
Comment (sdankel) [inline test/src/e2e_vm_tests/test_programs/should_fail/trait_cannot_find_in_scope_issue/test.toml:7]: Why did this change? We aren't running `fmt` on any test code are we?

File: swayfmt/src/utils/map/newline.rs [hunk 3]
```diff
@@ -45,59 +39,50 @@ fn is_new_line_in_rope(rope: &Rope, index: usize) -> bool {
...
     let mut in_sequence = false;
     let mut sequence_start = 0;
-    let os_new_line = NEW_LINE.chars().collect::<Vec<_>>();
-    while let Some((char_index, char)) = input_iter.next() {
-        let next_char = input_iter.peek().map(|input| input.1);
-        if matches!(char, '/' | ';' | '}')
-            && is_new_line_next_in_iter(&mut input_iter, &os_new_line)
-        {
+    let mut bytes_offset = 0;
+    while let Some(char) = input_iter.next() {
```
Comment (sdankel) [inline swayfmt/src/utils/map/newline.rs:51]: Let's move this to a separate PR so it's clear what bug this is fixing. It will be easier to review separately.

File: sway-lib-std/src/u256.sw [hunk 4]
```diff
@@ -293,28 +293,26 @@ impl U256 {
...
-                                                            (self.c == other.c && self.d < other.d)))) )
+        self.a < other.a || (self.a == other.a && (self.b < other.b || (self.b == other.b && (self.c < other.c || (self.c == other.c && self.d < other.d)))))
     }
 }
 
 #[test]
 fn test_u256_ord() {
     assert(U256::from((0, 0, 0, 1)) < U256::from((0, u64::max(), 0, 0)));
-    assert(!(U256::from((0, 0, 0, 1)) > U256::from((0, u64::max(), 0, 0))));
+    assert(
```
Comment (sdankel) [inline sway-lib-std/src/u256.sw:308]: It appears our default maximum line length is much shorter than rustfmt's. I can add 22 characters to this line before rustfmt will wrap it. Perhaps we should use the same as rustfmt.

File: sway-lib-std/src/u256.sw [hunk 5]
```diff
@@ -574,12 +572,14 @@ impl core::ops::Multiply for U256 {
...
                 let (overflow_of_b_to_a_3, b) = b.overflowing_add(overflow_of_c_to_b_2).into();
 
-                Self::from((
-                    self.b * other.c + result_b_d.upper + overflow_of_b_to_a_3 + overflow_of_b_to_a_2 + overflow_of_b_to_a_1 + overflow_of_b_to_a_0,
-                    b,
-                    c,
-                    result_d_d.lower,
-                ))
+                Self::from(
+                    (
```
Comment (sdankel) [inline sway-lib-std/src/u256.sw:576]: This change seems unnecessary. I think the formatter should leave it alone in this case.

File: sway-lib-std/src/array_conversions/u16.sw [hunk 6]
```diff
@@ -84,11 +84,11 @@ impl u16 {
         let output = [0_u8, 0_u8];
 
         asm(input: self, off: 0xFF, i: 0x8, output: output, r1) {
-            srl r1 input i;
-            sb output r1 i0;
+            srl  r1 input i;
+            sb   output r1 i0;
```
Comment (JoshuaBatty) [inline sway-lib-std/src/array_conversions/u16.sw:88]: I feel like we either need to align every column perfectly or just go with a single space between each token.

**Review by sdankel — DISMISSED** [review]

File: sway-lib-std/src/array_conversions/u64.sw [hunk 7]
```diff
@@ -207,21 +235,25 @@ impl u64 {
...
-            sll  r3 d l;
-            or   r1 r2 r3;
-            sll  r2 e k;
-            or   r3 r1 r2;
-            sll  r1 f j;
-            or   r2 r3 r1;
-            sll  r3 g i;
-            or   r1 r2 r3;
-            or   r2 r1 h;
+        asm(
```
Comment (JoshuaBatty) [inline sway-lib-std/src/array_conversions/u64.sw:238]: this looks a little odd... having them every argument on their own line also seems excessive..? What does rust-fmt do in this situation?

File: sway-lib-std/src/vec.sw [hunk 8]
```diff
@@ -3,7 +3,7 @@ library;
 
 use ::alloc::{alloc, realloc};
 use ::assert::assert;
-use ::option::Option::{self, *};
+use ::option::Option::{*, self};
```
Comment (JoshuaBatty) [inline sway-lib-std/src/vec.sw:6]: I feel like `self` should take priority over `*` for these..

File: sway-lib-std/src/array_conversions/b256.sw [hunk 9]
```diff
@@ -66,10 +70,30 @@ impl b256 {
...
+                bytes[16], bytes[17], bytes[18], bytes[19], bytes[20],
+                bytes[21], bytes[22], bytes[23],
+            ],
+        );
+        let d = u64::from_le_bytes(
+            [
+                bytes[24], bytes[25], bytes[26], bytes[27], bytes[28],
+                bytes[29], bytes[30], bytes[31],
+            ],
+        );
```
Comment (kayagokalp) [inline sway-lib-std/src/array_conversions/b256.sw:96]: This looks a little too long vertically:

image

This is the same piece code under `rustfmt`, might not be worth to block but we can discuss this one.

image

**Review by kayagokalp — DISMISSED** [review]
I think after fixing the extra long vertical stuff, this looks much much better. Huge improvement!

File: sway-lib-std/src/array_conversions/b256.sw (hunk 9 above)
Comment (kayagokalp) [reply sway-lib-std/src/array_conversions/b256.sw]: Great!

**Review by kayagokalp — APPROVED** [review]

**Review by JoshuaBatty — DISMISSED** [review]

**Review by bitzoic — APPROVED** [review]

**Review by kayagokalp — APPROVED** [review]
